--- FLIF-0.2.2/src/CMakeLists.txt.orig	2016-11-08 12:33:17.000000000 +0100
+++ FLIF-0.2.2/src/CMakeLists.txt	2016-12-03 22:07:35.902563666 +0100
@@ -95,6 +95,8 @@
     target_compile_definitions(flif_exe PRIVATE ${DEFINITIONS_FOR_ALL_TARGETS} STATIC_GETOPT ) # prevents dllexporting symbols for getopt
 endif()
 
+install(TARGETS flif_exe DESTINATION bin)
+
 # library
 
 add_library(flif_lib SHARED ${COMMON_SOURCES} ${FLIF_SRC_DIR}/flif-enc.cpp ${FLIF_SRC_DIR}/library/flif-interface.cpp)
@@ -105,8 +105,8 @@
 target_link_libraries(flif_lib ${PNG_LIBRARY} ${STATIC_LINKED_LIBS})
 target_link_libraries(flif_lib_dec ${PNG_LIBRARY} ${STATIC_LINKED_LIBS})
 
-set_target_properties(flif_lib PROPERTIES OUTPUT_NAME flif)
-set_target_properties(flif_lib_dec PROPERTIES OUTPUT_NAME flif_dec)
+set_target_properties(flif_lib PROPERTIES OUTPUT_NAME flif SOVERSION 0)
+set_target_properties(flif_lib_dec PROPERTIES OUTPUT_NAME flif_dec SOVERSION 0)
 
 target_compile_definitions(flif_lib PRIVATE ${DEFINITIONS_FOR_ALL_TARGETS} FLIF_BUILD_DLL )
 target_compile_definitions(flif_lib_dec PRIVATE ${DEFINITIONS_FOR_ALL_TARGETS} FLIF_BUILD_DLL DECODER_ONLY )
@@ -113,6 +115,13 @@
 target_include_directories(flif_lib PRIVATE ${FLIF_SRC_DIR}/../extern)
 target_include_directories(flif_lib_dec PRIVATE ${FLIF_SRC_DIR}/../extern)
 
+install(TARGETS flif_lib flif_lib_dec DESTINATION lib${LIB_SUFFIX})
+install(FILES ${FLIF_SRC_DIR}/library/flif.h
+              ${FLIF_SRC_DIR}/library/flif_common.h
+              ${FLIF_SRC_DIR}/library/flif_dec.h
+              ${FLIF_SRC_DIR}/library/flif_enc.h
+        DESTINATION include)
+
 # viewflif
 
 if(${SDL2_FOUND})
@@ -121,6 +130,7 @@
     target_link_libraries(viewflif flif_lib_dec ${SDL2_LIBRARY} ${SDL2MAIN_LIBRARY})
     target_include_directories(viewflif PRIVATE ${FLIF_SRC_DIR}/library ${SDL2_INCLUDE_DIR})
     target_compile_definitions(viewflif PRIVATE FLIF_USE_DLL)
+    install(TARGETS viewflif DESTINATION bin)
 endif()
 
 # test
